File: Lab5/isValid.java
public class isValid {
    public static boolean isValidBST(TreeNode root)
    {

        if(root == null)
        {
            return true;
        }
        if((root.left == null || root.left.val < root.val) && (root.right == null || root.right.val > root.val))
        {
            return isValidBST(root.left) && isValidBST(root.right);

        }
        return false;
    }
    public static void main(String[] args)
    {
        TreeNode root = new TreeNode(4);
            TreeNode three = root.left = new TreeNode(3);
            TreeNode eight = root.right = new TreeNode(8);
            TreeNode one = root.left.left = new TreeNode(1);
            TreeNode ten = root.left.right = new TreeNode(10);
            TreeNode five = root.right.left = new TreeNode(5);
            TreeNode nine = root.right.right = new TreeNode(9);

            boolean holder = isValidBST(root);
            System.out.println(holder);

    }
}

File: Lab5/isValid.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def isValidBST(root):
    if root is None:
        return True
    if (root.left is None or root.left.val < root.val) and (root.right is None or root.right.val > root.val):
        return isValidBST(root.left) and isValidBST(root.right)
    return False

if __name__ == "__main__" :
    root = TreeNode(4)
    three = root.left = TreeNode(3)
    eight = root.right = TreeNode(8)
    one = root.left.left = TreeNode(1)
    ten = root.left.right = TreeNode(10)
    five = root.right.left = TreeNode(5)
    nine = root.right.right = TreeNode(9)

    holder = isValidBST(root)
    print(holder)
